platform/base/core/src/main/java/com/peregrine/sitemap/impl/DefaultSiteMapExtractor.java

```diff
@@ -147,4 +156,25 @@ protected SiteMapUrlBuilder getUrlBuilder() {
         return urlBuilder;
     }
 
+    @Override
+    protected Page getProxy(final Resource page) {
+        try {
+            final ResourceResolver resourceResolver = page.getResourceResolver();
+            final Version version = resourceResolver
+                    .adaptTo(Session.class)
+                    .getWorkspace()
+                    .getVersionManager()
+                    .getVersionHistory(PerUtil.getJcrContent(page.getPath()))
+                    .getVersionByLabel(PUBLISHED_LABEL);
+            if (isNull(version)) {
+                return null;
+            }
+
+            final Node node = version.getFrozenNode();
+            return new Page(page, resourceResolver.getResource(node.getPath()));
+        } catch (final RepositoryException e) {
+            return null;
```

Comment:
@reggie7 - Please add appropriate log level and message.

Reply:
Looks good. In the future, you can use substitutions in the logger. I believe the method signatures supports up two substitutions.

```
logger.trace("Unable to grab the published version of path: {} ", path, e);
```

Reply:
Done.